core/unit_test/TestViewTypedefs.cpp

```diff
@@ -55,13 +55,16 @@ constexpr bool test_view_typedefs_impl() {
   static_assert(std::is_same_v<typename ViewType::data_type, DataType>);
   static_assert(std::is_same_v<typename ViewType::const_data_type, typename data_analysis<DataType>::const_data_type>);
   static_assert(std::is_same_v<typename ViewType::non_const_data_type, typename data_analysis<DataType>::non_const_data_type>);
-  
+
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
```

Comment:
Same question about deprecated warnings

Reply:
Good point - I think we do, https://cloud.cees.ornl.gov/jenkins-ci/job/Kokkos/job/PR-7360/9/pipeline-console/?selected-node=151
I think I need to add those macros.